gsy_framework/utils.py

```diff
@@ -585,3 +590,36 @@ def use_default_if_null(input_value: Union[int, str, float], default_value: Unio
                         ) -> Union[int, str, float]:
     """Return input_value if not null, else return provided default_value"""
     return input_value if input_value not in NULL_VALUES else default_value
+
+
+def _generate_time_slot_list(slot_length: timedelta, sim_duration: timedelta,
+                             start_date: datetime) -> List[datetime]:
+    return [start_date.add(minutes=slot_length.seconds / 60 * time_diff) for time_diff in
```

Comment:
A good optimisation would be for this to return a generator and not a list. That way the returned list would not remain in memory, but only single datetimes.